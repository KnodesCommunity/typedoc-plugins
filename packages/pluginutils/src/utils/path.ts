// eslint-disable-next-line no-restricted-imports -- Re-export normalized
import { dirname as _dirname, join as _join, normalize as _normalize, relative as _relative, resolve as _resolve, basename, extname, isAbsolute, parse } from 'path';

import { isString } from 'lodash';

import { normalizePath } from 'typedoc';

export const resolve = ( from: string, ...to: Array<string | null | undefined> ) => normalizePath( _resolve( from, ...to.filter( isString ) ) );
export const join = ( ...paths: Array<string | null | undefined> ) => normalizePath( _join( ...paths.filter( isString ) ) );
export const relative = ( from: string, to: string ) => normalizePath( _relative( from, to ) );
export const dirname = ( path: string ) => normalizePath( _dirname( path ) );
export const normalize = ( path: string ) => normalizePath( _normalize( path ) );
export { isAbsolute, parse, basename, extname };
