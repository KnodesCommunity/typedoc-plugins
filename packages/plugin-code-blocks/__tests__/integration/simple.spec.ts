import { resolve } from 'path';

import { describeDocsFile, formatHtml, runPluginBeforeAll } from '#plugintestbed';

import { formatExpanded, simpleJson } from '../helpers';

const SRC_CODE = simpleJson( '"Foo"', '"Bar"' );
const EXAMPLE_CODE = simpleJson( '"Hello"', '"World"' );
const INLINE_CODE = simpleJson( '"hello"', '"world"' );

const rootDir = resolve( __dirname, '../mock-fs/simple' );
process.chdir( rootDir );
runPluginBeforeAll( rootDir, resolve( __dirname, '../../src/index' ), { options: { gitRemote: 'FAKE' }} );
describe( '`index.html`', describeDocsFile( rootDir, 'index.html', it => {
	it( 'should have correct content', ( content, dom ) => {
		expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

		const panel = dom.window.document.querySelector( '.col-content > .tsd-panel.tsd-typography' );
		expect( panel!.children ).toHaveLength( 5 );
		expect( panel!.children[0].outerHTML ).toEqual( '<p>Hello world</p>' );
		expect( panel!.children[1].outerHTML ).toEqual( formatExpanded( 'Test inline codeblock from relative', SRC_CODE ) );
		expect( panel!.children[2].outerHTML ).toEqual( formatExpanded( 'Test inline codeblock from project root', EXAMPLE_CODE ) );
		expect( panel!.children[3].outerHTML ).toEqual( formatExpanded( 'Test inline codeblock from module root', EXAMPLE_CODE ) );
		expect( panel!.children[4].outerHTML ).toEqual( formatExpanded( 'example-inline.json', INLINE_CODE ) );
	} );
	it( 'should have constant content', content => {
		expect( formatHtml( content ) ).toMatchSnapshot();
	} );
} ) );
describe( 'Codeblock', () => {
	describe( 'Block tag', () => {
		describe( '`classes/BlockBlock.TestBlockExplicitProject.html`', describeDocsFile( rootDir, 'classes/BlockBlock.TestBlockExplicitProject.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './examples/example-test.json', EXAMPLE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/BlockBlock.TestBlockRel.html`', describeDocsFile( rootDir, 'classes/BlockBlock.TestBlockRel.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './src/codeblock/src-test.json', SRC_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/BlockBlock.TestBlockRelativeModule.html`', describeDocsFile( rootDir, 'classes/BlockBlock.TestBlockRelativeModule.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './examples/example-test.json', EXAMPLE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
	} );
	describe( 'Inline tag', () => {
		describe( '`classes/BlocInline.TestInlineExplicitProject.html`', describeDocsFile( rootDir, 'classes/BlockInline.TestInlineExplicitProject.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './examples/example-test.json', EXAMPLE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/BlocInline.TestInlineRel.html`', describeDocsFile( rootDir, 'classes/BlockInline.TestInlineRel.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './src/codeblock/src-test.json', SRC_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/BlocInline.TestInlineRelativeModule.html`', describeDocsFile( rootDir, 'classes/BlockInline.TestInlineRelativeModule.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './examples/example-test.json', EXAMPLE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/BlocInline.TestInlineExampleProject.html`', describeDocsFile( rootDir, 'classes/BlockInline.TestInlineExampleProject.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( './examples/example-test.json', EXAMPLE_CODE ) );

				const heading = dom.window.document.querySelector( '.tsd-panel > .tsd-comment > h3' );
				expect( codeblocks[0].previousElementSibling ).toBe( heading );
				expect( heading!.innerHTML ).toEqual( 'Example' );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
	} );
} );
describe( 'Inline Codeblock', () => {
	describe( 'Block tag', () => {
		describe( '`classes/InlineBlock.TestInlineBlock.html`', describeDocsFile( rootDir, 'classes/InlineBlock.TestInlineBlock.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( 'test.json', INLINE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
	} );
	describe( 'Inline tag', () => {
		describe( '`classes/InlineInline.TestInlineInline.html`', describeDocsFile( rootDir, 'classes/InlineInline.TestInlineInline.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( 'test.json', INLINE_CODE ) );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
		describe( '`classes/InlineInline.TestInlineExampleInline.html`', describeDocsFile( rootDir, 'classes/InlineInline.TestInlineExampleInline.html', it => {
			it( 'should have correct content', ( content, dom ) => {
				expect( content ).toMatch( /<link\s+rel="stylesheet"\s+href="([^"]*?\/)?assets\/code-blocks\.css"\s*\/>/ );

				const codeblocks = dom.window.document.querySelectorAll( '.code-block' );
				expect( codeblocks ).toHaveLength( 1 );
				expect( codeblocks[0].outerHTML ).toEqual( formatExpanded( 'test.json', INLINE_CODE ) );

				const heading = dom.window.document.querySelector( '.tsd-panel > .tsd-comment > h3' );
				expect( codeblocks[0].previousElementSibling ).toBe( heading );
				expect( heading!.innerHTML ).toEqual( 'Example' );
			} );
			it( 'should have constant content', content => {
				expect( formatHtml( content ) ).toMatchSnapshot();
			} );
		} ) );
	} );
} );
