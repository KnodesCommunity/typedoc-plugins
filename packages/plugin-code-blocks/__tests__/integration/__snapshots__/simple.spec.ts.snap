// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Real behavior should render correctly 1`] = `
"<!DOCTYPE html>
<html class=\\"default\\">

<head>
	<meta charSet=\\"utf-8\\" />
	<meta http-equiv=\\"x-ua-compatible\\" content=\\"IE=edge\\" />
	<title>@knodes/typedoc-plugin-code-blocks</title>
	<meta name=\\"description\\" content=\\"Documentation for @knodes/typedoc-plugin-code-blocks\\" />
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
	<link rel=\\"stylesheet\\" href=\\"assets/style.css\\" />
	<link rel=\\"stylesheet\\" href=\\"assets/highlight.css\\" />
	<script async src=\\"assets/search.js\\" id=\\"search-script\\"></script>
	<link rel=\\"stylesheet\\" href=\\"assets/code-blocks.css\\" />
</head>

<body>
	<script>
		document.body.classList.add(localStorage.getItem(\\"tsd-theme\\") || \\"os\\")
	</script>
	<header>
		<div class=\\"tsd-page-toolbar\\">
			<div class=\\"container\\">
				<div class=\\"table-wrap\\">
					<div class=\\"table-cell\\" id=\\"tsd-search\\" data-base=\\".\\">
						<div class=\\"field\\"><label for=\\"tsd-search-field\\" class=\\"tsd-widget search no-caption\\">Search</label><input type=\\"text\\" id=\\"tsd-search-field\\" /></div>
						<ul class=\\"results\\">
							<li class=\\"state loading\\">Preparing search index...</li>
							<li class=\\"state failure\\">The search index is not available</li>
						</ul><a href=\\"index.html\\" class=\\"title\\">@knodes/typedoc-plugin-code-blocks</a>
					</div>
					<div class=\\"table-cell\\" id=\\"tsd-widgets\\">
						<div id=\\"tsd-filter\\"><a href=\\"#\\" class=\\"tsd-widget options no-caption\\" data-toggle=\\"options\\">Options</a>
							<div class=\\"tsd-filter-group\\">
								<div class=\\"tsd-select\\" id=\\"tsd-filter-visibility\\"><span class=\\"tsd-select-label\\">All</span>
									<ul class=\\"tsd-select-list\\">
										<li data-value=\\"public\\">Public</li>
										<li data-value=\\"protected\\">Public/Protected</li>
										<li data-value=\\"private\\" class=\\"selected\\">All</li>
									</ul>
								</div> <input type=\\"checkbox\\" id=\\"tsd-filter-inherited\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-inherited\\">Inherited</label><input type=\\"checkbox\\" id=\\"tsd-filter-externals\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-externals\\">Externals</label>
							</div>
						</div><a href=\\"#\\" class=\\"tsd-widget menu no-caption\\" data-toggle=\\"menu\\">Menu</a>
					</div>
				</div>
			</div>
		</div>
		<div class=\\"tsd-page-title\\">
			<div class=\\"container\\">
				<h1>@knodes/typedoc-plugin-code-blocks</h1>
			</div>
		</div>
	</header>
	<div class=\\"container container-main\\">
		<div class=\\"row\\">
			<div class=\\"col-8 col-content\\">
				<section class=\\"tsd-panel-group tsd-index-group\\">
					<h2>Index</h2>
					<section class=\\"tsd-panel tsd-index-panel\\">
						<div class=\\"tsd-index-content\\">
							<section class=\\"tsd-index-section \\">
								<h3>Functions</h3>
								<ul class=\\"tsd-index-list\\">
									<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testNoPrefixImplicitInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />Implicit<wbr />In<wbr />Blocks</a></li>
									<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testNoPrefixInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />In<wbr />Blocks</a></li>
									<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testProjImplicitInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />Proj<wbr />Implicit<wbr />In<wbr />Blocks</a></li>
									<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testProjInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />Proj<wbr />In<wbr />Blocks</a></li>
									<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testRel\\" class=\\"tsd-kind-icon\\">test<wbr />Rel</a></li>
								</ul>
							</section>
						</div>
					</section>
				</section>
				<section class=\\"tsd-panel-group tsd-member-group \\">
					<h2>Functions</h2>
					<section class=\\"tsd-panel tsd-member tsd-kind-function\\"><a id=\\"testNoPrefixImplicitInBlocks\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\">test<wbr />No<wbr />Prefix<wbr />Implicit<wbr />In<wbr />Blocks<a href=\\"#testNoPrefixImplicitInBlocks\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<ul class=\\"tsd-signatures tsd-kind-function\\">
							<li class=\\"tsd-signature tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />Implicit<wbr />In<wbr />Blocks<span class=\\"tsd-signature-symbol\\">(</span><span class=\\"tsd-signature-symbol\\">)</span><span class=\\"tsd-signature-symbol\\">: </span><span class=\\"tsd-signature-type\\">number</span></li>
						</ul>
						<ul class=\\"tsd-descriptions\\">
							<li class=\\"tsd-description\\">
								<aside class=\\"tsd-sources\\">
									<ul>
										<li>Defined in test.ts:18</li>
									</ul>
								</aside>
								<div class=\\"tsd-comment tsd-typography\\">
									<div class=\\"lead\\">
										<p>A test code block for unprefixed path implicitly in blocks directory</p>
									</div>
									<div>
										<details class=\\"code-block\\" open>
											<summary>
												<p>From ./blocks/test.json</p>
											</summary>
											<pre><code class=\\"language-json\\"><span class=\\"hl-0\\">{</span><span class=\\"hl-1\\">&quot;Hello&quot;</span><span class=\\"hl-0\\">: </span><span class=\\"hl-2\\">&quot;World&quot;</span><span class=\\"hl-0\\">}</span>
</code></pre>
										</details>
									</div>
								</div>
								<h4 class=\\"tsd-returns-title\\">Returns <span class=\\"tsd-signature-type\\">number</span></h4>
							</li>
						</ul>
					</section>
					<section class=\\"tsd-panel tsd-member tsd-kind-function\\"><a id=\\"testNoPrefixInBlocks\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\">test<wbr />No<wbr />Prefix<wbr />In<wbr />Blocks<a href=\\"#testNoPrefixInBlocks\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<ul class=\\"tsd-signatures tsd-kind-function\\">
							<li class=\\"tsd-signature tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />In<wbr />Blocks<span class=\\"tsd-signature-symbol\\">(</span><span class=\\"tsd-signature-symbol\\">)</span><span class=\\"tsd-signature-symbol\\">: </span><span class=\\"tsd-signature-type\\">number</span></li>
						</ul>
						<ul class=\\"tsd-descriptions\\">
							<li class=\\"tsd-description\\">
								<aside class=\\"tsd-sources\\">
									<ul>
										<li>Defined in test.ts:25</li>
									</ul>
								</aside>
								<div class=\\"tsd-comment tsd-typography\\">
									<div class=\\"lead\\">
										<p>A test code block for unprefixed path in blocks directory</p>
									</div>
									<div>
										<details class=\\"code-block\\" open>
											<summary>
												<p>From ./blocks/test.json</p>
											</summary>
											<pre><code class=\\"language-json\\"><span class=\\"hl-0\\">{</span><span class=\\"hl-1\\">&quot;Hello&quot;</span><span class=\\"hl-0\\">: </span><span class=\\"hl-2\\">&quot;World&quot;</span><span class=\\"hl-0\\">}</span>
</code></pre>
										</details>
									</div>
								</div>
								<h4 class=\\"tsd-returns-title\\">Returns <span class=\\"tsd-signature-type\\">number</span></h4>
							</li>
						</ul>
					</section>
					<section class=\\"tsd-panel tsd-member tsd-kind-function\\"><a id=\\"testProjImplicitInBlocks\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\">test<wbr />Proj<wbr />Implicit<wbr />In<wbr />Blocks<a href=\\"#testProjImplicitInBlocks\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<ul class=\\"tsd-signatures tsd-kind-function\\">
							<li class=\\"tsd-signature tsd-kind-icon\\">test<wbr />Proj<wbr />Implicit<wbr />In<wbr />Blocks<span class=\\"tsd-signature-symbol\\">(</span><span class=\\"tsd-signature-symbol\\">)</span><span class=\\"tsd-signature-symbol\\">: </span><span class=\\"tsd-signature-type\\">number</span></li>
						</ul>
						<ul class=\\"tsd-descriptions\\">
							<li class=\\"tsd-description\\">
								<aside class=\\"tsd-sources\\">
									<ul>
										<li>Defined in test.ts:39</li>
									</ul>
								</aside>
								<div class=\\"tsd-comment tsd-typography\\">
									<div class=\\"lead\\">
										<p>A test code block for project path implicitly in blocks directory</p>
									</div>
									<div>
										<details class=\\"code-block\\" open>
											<summary>
												<p>From ./blocks/test.json</p>
											</summary>
											<pre><code class=\\"language-json\\"><span class=\\"hl-0\\">{</span><span class=\\"hl-1\\">&quot;Hello&quot;</span><span class=\\"hl-0\\">: </span><span class=\\"hl-2\\">&quot;World&quot;</span><span class=\\"hl-0\\">}</span>
</code></pre>
										</details>
									</div>
								</div>
								<h4 class=\\"tsd-returns-title\\">Returns <span class=\\"tsd-signature-type\\">number</span></h4>
							</li>
						</ul>
					</section>
					<section class=\\"tsd-panel tsd-member tsd-kind-function\\"><a id=\\"testProjInBlocks\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\">test<wbr />Proj<wbr />In<wbr />Blocks<a href=\\"#testProjInBlocks\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<ul class=\\"tsd-signatures tsd-kind-function\\">
							<li class=\\"tsd-signature tsd-kind-icon\\">test<wbr />Proj<wbr />In<wbr />Blocks<span class=\\"tsd-signature-symbol\\">(</span><span class=\\"tsd-signature-symbol\\">)</span><span class=\\"tsd-signature-symbol\\">: </span><span class=\\"tsd-signature-type\\">number</span></li>
						</ul>
						<ul class=\\"tsd-descriptions\\">
							<li class=\\"tsd-description\\">
								<aside class=\\"tsd-sources\\">
									<ul>
										<li>Defined in test.ts:32</li>
									</ul>
								</aside>
								<div class=\\"tsd-comment tsd-typography\\">
									<div class=\\"lead\\">
										<p>A test code block for project path in blocks directory</p>
									</div>
									<div>
										<details class=\\"code-block\\" open>
											<summary>
												<p>From ./blocks/test.json</p>
											</summary>
											<pre><code class=\\"language-json\\"><span class=\\"hl-0\\">{</span><span class=\\"hl-1\\">&quot;Hello&quot;</span><span class=\\"hl-0\\">: </span><span class=\\"hl-2\\">&quot;World&quot;</span><span class=\\"hl-0\\">}</span>
</code></pre>
										</details>
									</div>
								</div>
								<h4 class=\\"tsd-returns-title\\">Returns <span class=\\"tsd-signature-type\\">number</span></h4>
							</li>
						</ul>
					</section>
					<section class=\\"tsd-panel tsd-member tsd-kind-function\\"><a id=\\"testRel\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\">test<wbr />Rel<a href=\\"#testRel\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<ul class=\\"tsd-signatures tsd-kind-function\\">
							<li class=\\"tsd-signature tsd-kind-icon\\">test<wbr />Rel<span class=\\"tsd-signature-symbol\\">(</span><span class=\\"tsd-signature-symbol\\">)</span><span class=\\"tsd-signature-symbol\\">: </span><span class=\\"tsd-signature-type\\">number</span></li>
						</ul>
						<ul class=\\"tsd-descriptions\\">
							<li class=\\"tsd-description\\">
								<aside class=\\"tsd-sources\\">
									<ul>
										<li>Defined in test.ts:9</li>
									</ul>
								</aside>
								<div class=\\"tsd-comment tsd-typography\\">
									<div class=\\"lead\\">
										<p>A test code block for relative path from file</p>
									</div>
									<div>
										<details class=\\"code-block\\" open>
											<summary>
												<p>From ./src/test.json</p>
											</summary>
											<pre><code class=\\"language-json\\"><span class=\\"hl-0\\">{</span><span class=\\"hl-1\\">&quot;Foo&quot;</span><span class=\\"hl-0\\">: </span><span class=\\"hl-2\\">&quot;Bar&quot;</span><span class=\\"hl-0\\">}</span>
</code></pre>
										</details>
									</div>
								</div>
								<h4 class=\\"tsd-returns-title\\">Returns <span class=\\"tsd-signature-type\\">number</span></h4>
							</li>
						</ul>
					</section>
				</section>
			</div>
			<div class=\\"col-4 col-menu menu-sticky-wrap menu-highlight\\">
				<nav class=\\"tsd-navigation primary\\">
					<ul>
						<li class=\\"current\\"><a href=\\"modules.html\\">Exports</a></li>
					</ul>
				</nav>
				<nav class=\\"tsd-navigation secondary menu-sticky\\">
					<ul>
						<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testNoPrefixImplicitInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />Implicit<wbr />In<wbr />Blocks</a></li>
						<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testNoPrefixInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />No<wbr />Prefix<wbr />In<wbr />Blocks</a></li>
						<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testProjImplicitInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />Proj<wbr />Implicit<wbr />In<wbr />Blocks</a></li>
						<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testProjInBlocks\\" class=\\"tsd-kind-icon\\">test<wbr />Proj<wbr />In<wbr />Blocks</a></li>
						<li class=\\"tsd-kind-function\\"><a href=\\"modules.html#testRel\\" class=\\"tsd-kind-icon\\">test<wbr />Rel</a></li>
					</ul>
				</nav>
			</div>
		</div>
	</div>
	<footer class=\\"with-border-bottom\\">
		<div class=\\"container\\">
			<h2>Settings</h2>
			<p>Theme <select id=\\"theme\\">
					<option value=\\"os\\">OS</option>
					<option value=\\"light\\">Light</option>
					<option value=\\"dark\\">Dark</option>
				</select></p>
		</div>
	</footer>
	<div class=\\"container tsd-generator\\">
		<p>Generated using <a href=\\"https://typedoc.org/\\" target=\\"_blank\\">TypeDoc</a></p>
	</div>
	<div class=\\"overlay\\"></div>
	<script src=\\"assets/main.js\\"></script>
</body>

</html>"
`;
